### PR TITLE
Marketplace: Stop using the redirectTo parameter and mount the Marketplace Thank you page URL with the proper slugs on the Checkout page

### LINKED ISSUE #71675: Marketplace: Update the page redirect to Post-Purchase page with multiple plugins
After #70890 the Post Purchase page will accept multiple plugins, we should find a way to call this page passing multiple plugins.

### PR DESCRIPTION
**Pull request by WBerredo**
#### Proposed Changes

* Stop pass the `redirectTo` parameter
* Add the `plugin_slug` as extra property on the product - D97857-code
* Mount the Marketplace Thank you page URL with the slugs if a marketplace product is found on the cart

Screen Shot 2023-01-12 at 11 05 07

#### Testing Instructions
* Use the D97857-code change on your sandbox
* From a business or higher plan, go to a paid product page(`/plugins/{slug}/site`) and click on `Purchase and activate`
* Close the cart with the `X` on the top left of the screen and click on `Leave items`
* Go to another paid product page and click on `Purchase and activate`
* Complete the purchase
* You should see the Thank you page with all the purchased plugins

* From a below-than-business plan, repeat the steps above and you should see the same result.

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Fixes #71675

### CODE REVIEW

**Review by cpapazoglou — APPROVED** [review]
Tested this in an Atomic site with two marketplace plugins and in simple site with a business plan and a marketplace plugin. Works great!

File: client/my-sites/checkout/get-thank-you-page-url/index.ts [hunk 1]
```diff
@@ -516,6 +516,16 @@ function getFallbackDestination( {
...
 	}
 
+	const marketplaceProductSlugs =
+		cart?.products
+			?.filter( ( product ) => product?.extra?.is_marketplace_product )
+			?.map( ( product ) => product?.extra?.plugin_slug ) || [];
+
+	if ( marketplaceProductSlugs.length > 0 ) {
+		debug( 'site with marketplace products' );
+		return `/marketplace/thank-you/${ marketplaceProductSlugs.join( ',' ) }/${ siteSlug }`;
```
Comment (sirbrillig) [inline client/my-sites/checkout/get-thank-you-page-url/index.ts:526]: As described at the top of this file, could you add a test to cover this behavior change?

**Comment by sirbrillig** [comment]
Also, could you update the title of this PR to mention that this only (I think) affects marketplace products?

**Review by sirbrillig — APPROVED** [review]
Looks good to me!

**Pull request merged**